components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/impl/UserInfoJWTResponse.java

```diff
@@ -66,7 +77,57 @@ public String getResponseString(OAuth2TokenValidationResponseDTO tokenResponse)
 
         JWTClaimsSet jwtClaimsSet = new JWTClaimsSet();
         jwtClaimsSet.setAllClaims(claims);
-        return new PlainJWT(jwtClaimsSet).serialize();
+
+        String sigAlg = OAuthServerConfiguration.getInstance().getUserInfoJWTSignatureAlgorithm();
+        if (StringUtils.isNotBlank(sigAlg)) {
+            try {
+                signatureAlgorithm = OAuth2Util.mapSignatureAlgorithmForJWSAlgorithm(sigAlg);
+            } catch (IdentityOAuth2Exception e) {
+                throw new UserInfoEndpointException("Provided signature algorithm : " + sigAlg + " is not supported.",
+                        e);
+            }
+        }
+
+        if (JWSAlgorithm.NONE.getName().equals(signatureAlgorithm.getName())) {
+            if (log.isDebugEnabled()) {
+                log.debug("User Info JWT Signature algorithm is not defined. Returning unsigned JWT.");
+            }
+            return new PlainJWT(jwtClaimsSet).serialize();
+        }
+
+        boolean isJWTSignedWithSPKey = OAuthServerConfiguration.getInstance().isJWTSignedWithSPKey();
```

Comment:
Need to add this configuration to default identity.xml at least commented.

Reply:
have,
isJWTSignedWithSPKey=true
as commented..

Having this commented is important since it required for backward compatibility.

Reply:
this is already added with,
isJWTSignedWithSPKey=false
uncommented.

Ref commit: https://github.com/wso2/carbon-identity-framework/commit/f87b6d2025ab346b2f06fd4ab3aa2ddb63ed0ffc